### PR TITLE
[Feature] Support statistics system table for paimon

### PR DESCRIPTION
**Pull request by xuzifu666**
Support table statistics system table for paimon

Linked issue: https://github.com/apache/paimon/issues/3169

### CODE REVIEW

**Comment by xuzifu666** [comment]
@JingsongLi Add a statistic table for user query， PTAL

**Comment by xuzifu666** [comment]
> @xuzifu666 Thanks for the contribution, can you add test cases?
 OK，add ut latter @Zouxxyy

**Comment by xuzifu666** [comment]
With the pr user can query like select * from T$statistic and result as

File: paimon-core/src/main/java/org/apache/paimon/table/system/StatisticTable.java [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+import java.util.Objects;
+
+import static org.apache.paimon.catalog.Catalog.SYSTEM_TABLE_SPLITTER;
+
+/** A {@link Table} for showing statistic of table. */
+public class StatisticTable implements ReadonlyTable {
+
+    private static final long serialVersionUID = 1L;
+
+    public static final String STATISTIC = "statistic";
```
Comment (Zouxxyy) [inline paimon-core/src/main/java/org/apache/paimon/table/system/StatisticTable.java:63]: STATISTICS

File: paimon-core/src/main/java/org/apache/paimon/table/system/StatisticTable.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+    public static final RowType TABLE_TYPE =
+            new RowType(
+                    Arrays.asList(
+                            new DataField(0, "snapshot_id", new BigIntType(false)),
+                            new DataField(1, "schema_id", new BigIntType(false)),
+                            new DataField(
+                                    2, "content_json", SerializationUtils.newStringType(false))));
```
Comment (Zouxxyy) [inline paimon-core/src/main/java/org/apache/paimon/table/system/StatisticTable.java:71]: "statistic"

File: paimon-core/src/main/java/org/apache/paimon/table/system/StatisticTable.java [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    }
+
+    @Override
+    public InnerTableScan newScan() {
+        return new StatisticTable.StatisticScan();
+    }
+
+    @Override
+    public InnerTableRead newRead() {
+        return new StatisticTable.StatisticRead(fileIO, dataTable);
```
Comment (Zouxxyy) [inline paimon-core/src/main/java/org/apache/paimon/table/system/StatisticTable.java:106]: return new StatisticRead(fileIO, dataTable);

File: paimon-core/src/main/java/org/apache/paimon/table/system/StatisticTable.java [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+                                rows, row -> ProjectedRow.from(projection).replaceRow(row));
+            }
+            return new IteratorRecordReader<>(rows);
+        }
+
+        private InternalRow toRow(Statistics statistics) {
+            return GenericRow.of(
+                    statistics.snapshotId(),
+                    statistics.schemaId(),
+                    BinaryString.fromString(statistics.toJson()));
```
Comment (Zouxxyy) [inline paimon-core/src/main/java/org/apache/paimon/table/system/StatisticTable.java:221]: see https://github.com/apache/paimon/pull/2836/files#r1475434831

File: paimon-core/src/main/java/org/apache/paimon/table/system/StatisticTable.java (hunk 3 above)
Comment (xuzifu666) [reply paimon-core/src/main/java/org/apache/paimon/table/system/StatisticTable.java]: addressed

File: paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/sql/AnalyzeTableTestBase.scala [hunk 5]
```diff
@@ -50,6 +50,23 @@ abstract class AnalyzeTableTestBase extends PaimonSparkTestBase {
...
+                 |""".stripMargin)
+
+    spark.sql(s"INSERT INTO T VALUES ('1', 'a', 1, 1)")
+    spark.sql(s"INSERT INTO T VALUES ('2', 'aaa', 1, 2)")
+
+    spark.sql(s"ANALYZE TABLE T COMPUTE STATISTICS")
+
+    val table = "`T$statistics`"
+    val df = spark.sql(s"select * from $table")
+    Assertions.assertEquals(df.collect().size, 1)
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/sql/AnalyzeTableTestBase.scala:67]: Why just assert size?

File: paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/sql/AnalyzeTableTestBase.scala [hunk 6]
```diff
@@ -50,6 +50,23 @@ abstract class AnalyzeTableTestBase extends PaimonSparkTestBase {
...
+                 |USING PAIMON
+                 |TBLPROPERTIES ('primary-key'='id')
+                 |""".stripMargin)
+
+    spark.sql(s"INSERT INTO T VALUES ('1', 'a', 1, 1)")
+    spark.sql(s"INSERT INTO T VALUES ('2', 'aaa', 1, 2)")
+
+    spark.sql(s"ANALYZE TABLE T COMPUTE STATISTICS")
+
+    val table = "`T$statistics`"
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/sql/AnalyzeTableTestBase.scala:65]: Just inline this

Comment (xuzifu666) [reply paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/sql/AnalyzeTableTestBase.scala]: addressed

File: paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/sql/AnalyzeTableTestBase.scala (hunk 5 above)
Comment (xuzifu666) [reply paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/sql/AnalyzeTableTestBase.scala:66]: Had assert details in ut， Thanks @JingsongLi

**Comment by JingsongLi** [comment]
+1

**Pull request merged**